src/query/deps.go

```diff
@@ -37,19 +37,21 @@ func deps(out io.Writer, state *core.BuildState, target *core.BuildTarget, done
 			continue // target is filtered out
 		}
 		done[dep] = true
-		if hidden || !dep.HasParent() {
-			// dep is to be printed; either we're printing hidden deps or it has no parent (i.e. is not hidden)
-			if formatdot {
-				printTargetDot(out, dep, target)
+		for _, l := range dep.ProvideFor(target) {
+			if dep := state.Graph.TargetOrDie(l); hidden || !dep.HasParent() {
```

Comment:
So this just turns the target into a label, and the label back into a target?

Reply:
I see now, I needed to read [the docs on require/provide](https://please.build/require_provide.html), I was reading provide in a generic sense, rather than please-specific sense.
Makes sense!

Reply:
Yeah, it resolves require/provide for that target then keeps going with that